cmd/minikube/cmd/start_flags.go

```diff
@@ -494,6 +509,114 @@ func getNetwork(driverName string, options *run.CommandOptions) string {
+func normalizeAndValidateIPFamily(cc *config.ClusterConfig) {
+	fam := strings.ToLower(strings.TrimSpace(cc.KubernetesConfig.IPFamily))
+	switch fam {
+	case "", "ipv4", "ipv6", "dual":
+		// ok
+	default:
+		exit.Message(reason.Usage,
+			"Invalid --ip-family {{.fam}}. Must be one of: ipv4, ipv6, dual.",
+			out.V{"fam": cc.KubernetesConfig.IPFamily})
+	}
+
+	// Default to ipv4 to keep existing behavior for old configs.
+	if fam == "" {
+		fam = "ipv4"
+		cc.KubernetesConfig.IPFamily = fam
+	}
+
+	// IPv6-capable families: ensure v6 defaults.
+	if fam != "ipv4" {
+		if cc.KubernetesConfig.ServiceCIDRv6 == "" {
+			cc.KubernetesConfig.ServiceCIDRv6 = constants.DefaultServiceCIDRv6
+		}
+		if cc.KubernetesConfig.PodCIDRv6 == "" {
+			cc.KubernetesConfig.PodCIDRv6 = constants.DefaultPodCIDRv6
+		}
+	}
+
+	// For ipv4 or dual-stack, keep the usual IPv4 pod CIDR default if none was set.
+	if fam != "ipv6" && cc.KubernetesConfig.PodCIDR == "" {
+		cc.KubernetesConfig.PodCIDR = cni.DefaultPodCIDR
+	}
+
+	// --- CIDR / address validation ---
+
+	// Helper: ensure a CIDR is v4 or v6 as expected.
+	checkCIDR := func(flag string, cidr string, wantV6 bool) {
+		ip, _, err := net.ParseCIDR(cidr)
+		if err != nil {
+			exit.Message(reason.Usage, "{{.flag}} must be a valid CIDR: {{.e}}",
+				out.V{"flag": flag, "e": err})
+		}
+		isV4 := ip.To4() != nil
+		if wantV6 && isV4 {
+			exit.Message(reason.Usage, "{{.flag}} must be a valid IPv6 CIDR: {{.e}}",
+				out.V{"flag": flag, "e": err})
+		}
+		if !wantV6 && !isV4 {
+			exit.Message(reason.Usage, "{{.flag}} must be a valid IPv4 CIDR: {{.e}}",
+				out.V{"flag": flag, "e": err})
+		}
+	}
+
+	// IPv6-only CIDR flags.
+	if cidr := cc.Subnetv6; cidr != "" {
+		checkCIDR("--subnet-v6", cidr, true)
+	}
+	if cidr := cc.KubernetesConfig.ServiceCIDRv6; cidr != "" {
+		checkCIDR("--service-cluster-ip-range-v6", cidr, true)
+	}
+	if cidr := cc.KubernetesConfig.PodCIDRv6; cidr != "" {
+		checkCIDR("--pod-cidr-v6", cidr, true)
+	}
+
+	// IPv4-only pod CIDR.
+	if cidr := cc.KubernetesConfig.PodCIDR; cidr != "" {
+		checkCIDR("--pod-cidr", cidr, false)
+	}
+
+	// validate static IPv6 if provided
+	if s := cc.StaticIPv6; s != "" {
+		ip := net.ParseIP(s)
+		if ip == nil || ip.To4() != nil {
+			exit.Message(reason.Usage, "--static-ipv6 must be a valid IPv6 address")
+		}
+	}
+
+	// Docker driver guardrails: Linux daemon for IPv6, warn on Desktop.
+	if driver.IsDocker(cc.Driver) && fam != "ipv4" {
+		// Desktop vs Linux daemon hint (we can't reliably detect IPv6 enabled here).
+		si, err := oci.CachedDaemonInfo(cc.Driver)
+		if err != nil {
+			si, err = oci.DaemonInfo(cc.Driver)
+			if err != nil {
+				exit.Message(reason.Usage, "Failed to query Docker daemon info: {{.e}}", out.V{"e": err})
+			}
+		}
+
+		// On non-Linux hosts we assume Docker Desktop; on Linux it's a native Engine
+		// unless DockerOS explicitly says "Docker Desktop".
+		isLinuxDaemon := runtime.GOOS == "linux" && si.DockerOS != "Docker Desktop"
```

Comment:
I'm wondering about this check. Should it use daemon OS, not client OS? Can you please explain your reasoning a bit more?

Does this mean running docker inside WSL2 without docker desktop won't work?